lighthouse-cli/test/cli/run-test.js

```diff
@@ -57,6 +64,7 @@ describe('flag coercing', () => {
 
 describe('saveResults', () => {
   it('will quit early if we\'re in gather mode', async () => {
+    // @ts-ignore - testing a thin part of the interface
```

Comment:
> Which isn't much better than the current code. wdyt?

Mocking types is going to come up as an issue like this a whole lot when/if we convert `core/test/` over to type checking. I think we do want to default to as little `// @ts-ignore` as possible because the compiler ends up blind there. Casting/assertions have the same problem, of course, but at least it's limited to the single variable, and tsc will catch the small case of if the types become mutually incompatible.

The ideal solution is the mocks become good enough to properly match the types, but failing that for this case, we can also do:
```js
const result = await run.saveResults(
  /** @type {LH.RunnerResult} */ ({}),
  /** @type {LH.CliFlags} */ ({gatherMode: true}));
```
(I think the errors you saw were if you don't include the `@type`?)

It looks a little sloppy, but it's also an expression of how narrowly the test is targeting the internals of the current `saveResults` implementation, so it seems somewhat appropriate :) WDYT?

Reply:
Oh yes, that code works. I must have mistyped the jsdoc comment.